lms/djangoapps/experiments/views_custom.py

```diff
@@ -0,0 +1,166 @@
+# .. feature_toggle_default: False
+# .. feature_toggle_description: Toggle mobile upsell enabled
+# .. feature_toggle_category: experiments
+# .. feature_toggle_use_cases: monitored_rollout
+# .. feature_toggle_creation_date: 2019-09-05
+# .. feature_toggle_expiration_date: None
+# .. feature_toggle_warnings: None
+# .. feature_toggle_tickets: REV-934
+# .. feature_toggle_status: supported
+MOBILE_UPSELL_FLAG = WaffleFlag(
+    waffle_namespace=WaffleFlagNamespace(name=u'experiments'),
+    flag_name=u'mobile_upsell_rev934',
+    flag_undefined_default=False
+)
+MOBILE_UPSELL_EXPERIMENT = 'mobile_upsell_experiment'
+
+
+class Rev934(DeveloperErrorViewMixin, APIView):
+    """
+    **Use Cases**
+
+        Request upsell information for mobile app users
+
+    **Example Requests**
+
+        GET /api/experiments/v0/custom/REV-934/?course_id={course_key_string}
+
+    **Response Values**
+
+        Body consists of the following fields:
+            show_upsell:
+                whether to show upsell in the moble app in this case
+            price:
+                (optional) the price to show if show_upsell is true
+            basket_url:
+                (optional) the url to the checkout page with the course's sku if show_upsell is true
+            upsell_flag:
+                (optional) false if the upsell flag is off, not present otherwise
+
+        Response:
+            {
+            "show_upsell": true,
+            "price": "$199",
+            "basket_url": "https://ecommerce.edx.org/basket/add?sku=abcdef"
+            }
+
+    **Parameters:**
+
+        course_key_string:
+            The course key that may be upsold
+
+    **Returns**
+
+        * 200 on success with above fields.
+        * 401 if there is no user signed in.
+
+        Example response:
+        {
+            "show_upsell": true,
+            "price": "$199",
+            "basket_url": "https://ecommerce.edx.org/basket/add?sku=abcdef"
+        }
+    """
+    # http://localhost:18000/api/experiments/v0/custom/REV-934/?course_id=course-v1:edX+DemoX+Demo_Course
+
+    authentication_classes = (
+        JwtAuthentication,
+        OAuth2AuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
+    permission_classes = (ApiKeyHeaderPermissionIsAuthenticated,)
+
+    @method_decorator(ensure_csrf_cookie_cross_domain)
+    def get(self, request):
+        """
+        Return the if the course should be upsold in the mobile app, if the user has appropriate permissions.
+        """
+        if not MOBILE_UPSELL_FLAG.is_enabled():
+            return Response({
+                'show_upsell': False,
+                'upsell_flag': False,
+            })
+
+        if 'course_id' not in request.GET:
+            return Response({
+                'show_upsell': False,
+            })
+
+        course_id = request.GET.get('course_id').replace(' ', '+')  # HACK: the url decoding converts plus to space; put them back
```

Comment:
Perhaps we can ask in the Django channel if someone knows a less hacky way to do this?